src/RunTests/RunVSTestTask.cs

```diff
@@ -0,0 +1,443 @@
+                    {
+                        isCollectCodeCoverageEnabled = true;
+                    }
+
+                    allArgs.Add("--collect:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(arg));
+                }
+            }
+
+            if (isCollectCodeCoverageEnabled || isRunSettingsEnabled)
+            {
+                // Pass TraceDataCollector path to vstest.console as TestAdapterPath if --collect "Code Coverage"
+                // or --settings (User can enable code coverage from runsettings) option given.
+                // Not parsing the runsettings for two reason:
+                //    1. To keep no knowledge of runsettings structure in VSTestTask.
+                //    2. Impact of adding adapter path always is minimal. (worst case: loads additional data collector assembly in datacollector process.)
+                // This is required due to currently trace datacollector not ships with dotnet sdk, can be remove once we have
+                // go code coverage x-plat.
+                if (!string.IsNullOrEmpty(VSTestTraceDataCollectorDirectoryPath))
+                {
+                    allArgs.Add("--testAdapterPath:" +
+                                ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(
+                                    VSTestTraceDataCollectorDirectoryPath));
+                }
+            }
+
+            if (!string.IsNullOrEmpty(VSTestNoLogo))
+            {
+                allArgs.Add("--nologo");
+            }
+
+            if (!string.IsNullOrEmpty(VSTestArtifactsProcessingMode) && VSTestArtifactsProcessingMode.Equals("collect", StringComparison.OrdinalIgnoreCase))
+            {
+                allArgs.Add("--artifactsProcessingMode-collect");
+            }
+
+            if (!string.IsNullOrEmpty(VSTestSessionCorrelationId))
+            {
+                allArgs.Add("--testSessionCorrelationId:" + ArgumentEscaper.HandleEscapeSequenceInArgForProcessStart(VSTestSessionCorrelationId));
+            }
+
+            return allArgs;
+        }
+
+        private Task<int> ExecuteTest()
+        {
+#if NET6_0_OR_GREATER
+            string packagePath = $@"{NugetPath}\packages\microsoft.testplatform\{VSTestRunnerVersion}\tools\net6.0\Common7\IDE\Extensions\TestPlatform\";
+#else
+            string packagePath = $@"{NugetPath}\packages\microsoft.testplatform\{VSTestRunnerVersion}\tools\net462\Common7\IDE\Extensions\TestPlatform\";
+#endif
+            var processInfo = new ProcessStartInfo
+            {
+                FileName = $"{packagePath}vstest.console.exe",
+                Arguments = string.Join(" ", CreateArgument()),
+                UseShellExecute = false,
+            };
+
+            using (var activeProcess = new Process { StartInfo = processInfo })
+            {
+                activeProcess.Start();
+                activeProcess.WaitForExit();
+                return Task.FromResult(activeProcess.ExitCode);
```

Comment:
Since you're using `Task.FromResult`, no need to make this method return a task at all